### ISSUE TITLE
shaka - It is not possible to get WBTC price in Ethereum mainnet

### ISSUE THREAD

**Issue by sherlock-admin** [sherlock-audit/2023-05-ironbank-judging#263]
shaka

medium

# It is not possible to get WBTC price in Ethereum mainnet

## Summary

It is not possible to get WBTC price in Ethereum mainnet.

## Vulnerability Detail

WBTC is listed in the README as one of the supported assets. However, with the current implementation of `OraclePrice:getPrice()` it is not possible to get the price of WBTC in Ethereum mainnet, as there is no Chainlink WBTC/USD price feed.

## Impact

In the best-case scenario, WBTC market will not be available. 
In the worst-case scenario, BTC/USD price feed will be used. Given that the price of WBTC can be significantly different from the price of BTC, this can lead to the value of WBTC being over or underestimated, which can lead to arbitrage opportunities.

## Code Snippet

https://github.com/sherlock-audit/2023-05-ironbank/blob/main/ib-v2/src/protocol/oracle/PriceOracle.sol#L42-L57

## Tool used

Manual review

## Recommendation

When `asset` is WBTC, query both BTC/USD and WBTC/BTC price feeds and calculate the WBTC/USD price.

**Comment by shaka0x** [comment]
Escalate for 10 USDC

This should not be a duplicate of https://github.com/sherlock-audit/2023-05-ironbank-judging/issues/132. The motivation of that issues is that calculating the WBTC/USD price as the combination of the WBTC/BTC and BTC/USD feeds is not a good approach and there is a better alternative. The motivation of this issue is that the calculation of WBTC/USD price is NOT done with combination of the WBTC/BTC and BTC/USD feeds, although it should. In fact, in the sponsor's comment, the issues are addressed in separate points https://github.com/sherlock-audit/2023-05-ironbank-judging/issues/132#issuecomment-1598256777.

As for the sponsor`s comment:
```
We forgot to add BTC as quote: This is our oversight and we will fix this.
```
The sponsor is confirming that this needs to be fixed. I don't think that stating that it was an oversight is a reason for invalidating the issue, the same way that saying that not adding an `onlyOwner` modifier was an oversight should not invalidate an issue related with access control.

```
However, this issue will be found when we launch since there is no WBTC/USD price feed out there.
```
We cannot be sure that at the moment of the deployment the protocol team would not realize and the BTC/USD price feed would be used instead. In any case, for the protocol to be able to be deployed in a safe way, the code needs to be modified with the suggested recommendation.

**Comment by 0xffff11** [comment]
I see the issue as a duplicate, the difference that I see is that you recommendation is a bit better than the other duplicates. I still would consider it a duplicate if there is no other big difference

**Comment by ibsunhub** [comment]
fix: https://github.com/ibdotxyz/ib-v2/pull/52

**Comment by hrishibhat** [comment]
Result:
Invalid
Unique
This is not a duplicate of #132 
But this is a non-issue: 
As this is already mentioned in the docs. 
> There are some tokens that we cannot find the corresponding price feeds on Chainlink. Therefore, we use price feed alternative instead.
> For WBTC, we use the combination of two price feeds WBTC/BTC and BTC/USD, to get the WBTC price in USD quote as the source.

> WBTC/BTC: 0xfdFD9C85aD200c506Cf9e21F1FD8dd01932FBB23
> BTC/USD: 0xF4030086522a5bEEa4988F8cA5B36dbC97BeE88c

**Comment by sherlock-admin2** [comment]
Escalations have been resolved successfully!

Escalation status:
- [shaka0x](https://github.com/sherlock-audit/2023-05-ironbank-judging/issues/263/#issuecomment-1606197216): accepted

**Comment by IAm0x52** [comment]
Fix looks good. Same fix as #132
